config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-main-upgrades.yaml

```diff
@@ -25,13 +25,13 @@ periodics:
         - "./scripts/ci-e2e.sh"
       env:
         - name: KUBERNETES_VERSION_UPGRADE_FROM
-          value: "stable-1.22"
-        - name: KUBERNETES_VERSION_UPGRADE_TO
           value: "stable-1.23"
+        - name: KUBERNETES_VERSION_UPGRADE_TO
+          value: "stable-1.24"
         - name: ETCD_VERSION_UPGRADE_TO
-          value: "3.5.3-0"
+          value: "3.5.6-0"
         - name: COREDNS_VERSION_UPGRADE_TO
-          value: "v1.8.6"
+          value: "v1.9.3"
```

Comment:
tried to see if there is any support matrix for coreDNS / k8s versions but couldn't find one https://github.com/coredns/coredns, so I'm assuming this should be fine? CAPI doesn't use the same coreDNS version everywhere https://github.com/kubernetes/test-infra/commit/3107bae4e798ef5f3c949d7b4e2d3a6afc6b497e#diff-bdf74ee3955b26f3fc158ef2e0da0773659c4083e071a2af0d40e4e87a4a2da4R270